client/py/synnax/io/tdms.py

```diff
@@ -0,0 +1,144 @@
+        if self._channels is not None:
+            return self._channels
+        
+        self._channels: list[ChannelMeta] = list()
+        with TdmsFile.open(self._path) as tdms_file:
+            for group in tdms_file.groups():
+                for channel in group.channels():
+                    self._channels.append(ChannelMeta(name=channel.name, meta_data=dict()))
+                    
+        return self._channels
+    
+    def set_chunk_size(self, chunk_size: int):
+        self.chunk_size = chunk_size
+    
+    def set_keys(self, keys: list[str]):
+        self.channel_keys = set(keys)
+    
+    def set_keys_from_file(self) -> set[str]:
+        self.channel_keys = set([ch.name for ch in self.channels()])
+        return self.channel_keys
+
+    @classmethod
+    def type(cls) -> ReaderType:
+        """:returns : the type of reader."""
+        return ReaderType.Column
+
+    def path(self) -> Path:
+        """:returns: the path to the file."""
+        return self._path
+
+    def nsamples(self) -> int:
+        """:returns: the number of samples in the file."""
+        return self.chunk_size * self.n_chunks * len(self.channels())
+
+    def seek_first(self):
+        """Seeks the reader to the first sample in the file."""
+        self._current_chunk = 0
+    
+    def read(self, *keys: str) -> pd.DataFrame:
+        """Reads a dataframe with chunk size * number of columns samples. The returned
+        dataframe contains columns for each key in keys.
+        """ 
+        # if we already read everything, return an empty chunk
+        if self._current_chunk >= self.n_chunks:
+            return pd.DataFrame()
+        
+        # if keys is empty, use default keys
+        keys: set[str] = self.channel_keys if (len(keys) == 0) else set(keys)
+        
+        # https://nptdms.readthedocs.io/en/stable/reading.html
+        # https://github.com/adamreeve/npTDMS/issues/263
+        # https://nptdms.readthedocs.io/en/stable/reading.html
+        data = dict()
+        with TdmsFile.open(self._path) as tdms_file:            
+            for group in tdms_file.groups():  
+                for channel in group.channels():
+                    if channel.name in keys:
+                        data[channel.name] = channel[self._current_chunk*self.chunk_size:
+                                                    (self._current_chunk + 1)*self.chunk_size]
+            self._current_chunk += 1
+        
+        return pd.DataFrame(data)
+
+    def __iter__(self) -> Iterator[pd.DataFrame]:
+        return (self.read() for _ in range(self.n_chunks))
+    
+    @property
+    def n_chunks(self) -> int:
+        """Returns number of chunks in a channel.
+        Assumes every channel has the same number of chunks
+        """
+        if self._n_chunks is None:
+            # assume every channel has the same number of data
+            with TdmsFile.open(self._path) as tdms_file:
```

Comment:
Do we need to open this handle multiple times? I guess the overhead isn't huge, but I do like the thought of opening the file once and keeping a handle as a class property. Reduces the number of file descriptors.